src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -227,21 +260,24 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
             const auto& else_body_results = else_body->get_results();
             const auto& then_out_desc = if_op->get_output_descriptions(ov::op::v8::If::THEN_BODY_INDEX);
             const auto& else_out_desc = if_op->get_output_descriptions(ov::op::v8::If::ELSE_BODY_INDEX);
-            for (const auto& out_desc : then_out_desc) {
+
+            for (auto& out_desc : then_out_desc) {
                 const auto& out_indx = out_desc->m_output_index;
                 const auto& body_indx = out_desc->m_body_value_index;
-                then_body_results[body_indx]->get_input_tensor(0).m_partial_shape =
-                    if_op->get_output_partial_shape(out_indx);
-                then_body_results[body_indx]->get_input_tensor(0).m_element_type =
-                    if_op->get_output_element_type(out_indx);
+
+                set_source_output_type_shape(*then_body_results[body_indx],
+                                             if_op->get_output_element_type(out_indx),
+                                             if_op->get_output_partial_shape(out_indx),
+                                             0);
             }
-            for (const auto& out_desc : else_out_desc) {
+
+            for (auto& out_desc : else_out_desc) {
```

Comment:
Can't `out_desc` stay `const`?

Reply:
It is const because `else_out_desc` is const but will revert it.